evennia/contrib/aisystem/README.md

```diff
@@ -0,0 +1,661 @@
+SUCCESS when no enemies are sighted.
+
+`Succeeder` - this node returns SUCCESS both when its child returns FAILURE and
+when its child returns SUCCESS. It otherwise returns its child's status.
+
+`Failer` - this node returns FAILURE both when its child returns FAILURE and
+when its child returns SUCCESS. It otherwise returns its child's status.
+
+`Repeater` - each time it is run, this node runs its child multiple times, up
+to the number of times specified by its `repeats` property, while the child is
+returning SUCCESS or FAILURE. Afterwards, the Repeater returns SUCCESS. If at
+any point the child returns ERROR or RUNNING, the repeater immediately returns
+that status. Each node instance keeps track of the current number of repeats
+via its 'ticks' entry.
+
+`Limiter` - each time it is run, a given instance of this node increments its
+'ticks' entry by one, up to the point where that counter reaches the value of
+the node's `repeats` property. Before reaching this threshold, the node simply
+returns its child's return status. After reaching this threshold, however, it
+returns FAILURE without ever running its child. The Limiter node can be used
+for only allowing its subtree to be run a limited number of times. Keep in mind
+that you can reset its 'ticks' entry from other nodes that you code yourself,
+so the node's limiting effect need not be a permanent one.
+
+`Allocator` - an allocator only runs its child (and returns that child's return
+status) when none of the resources it requires are claimed. When one or more of
+the needed resources is claimed, it returns RUNNING without running its child.
+
+Note to coders: resources are entries in the node's `resources` property. They
+are of the type _string_. The 'globals' entry of a given blackboard has a
+'resources' entry that is a dictionary storing such strings as keys. If no key
+in this dictionary matches the name of a string entry in the node's `resources`
+property, or if such a key exists but is set to None, False, or an empty
+container (i.e. it returns False when converted to a _bool_), then the resource
+described by this string entry is considered freed. Otherwise, it is not
+considered freed and will prevent allocator nodes that rely on it from running
+their children until some bit of code frees it.
+
+`EchoDecorator` - the EchoDecorator sends a message to the room and runs its
+child, then returns that child's return status. It stores this message in the
+node's `msg` property and a given node instance's 'msg' entry. If no message is
+provided (this is the default scenario), the EchoDecorator returns FAILURE
+without sending a message or running its child. Note that this node may only be
+used when its agent is an AIObject, not an AIScript (see the section on setting
+up the AI system for more information.)
+
+#### Blackboards
+
+Because it would be inefficient to store an instance of a tree on each agent
```

Comment:
Is "blackboard" really a standard name for this type of agent storage? At any rate it would be good for understanding to get a visual metaphor explaining why it's called "blackboard". 
It might also be useful to learn what type of entity this is - an object stored on the Character, a list, dict?